frontend/src/components/Button/Button.tsx

```diff
@@ -0,0 +1,43 @@
+import { ReactNode, MouseEvent } from 'react';
+import { StyledButton } from './Button.styled';
+import { NativeButtonProps } from 'antd/lib/button/button';
+import 'antd/dist/antd.css';
+
+interface ButtonProps extends NativeButtonProps {
+  children: ReactNode;
+  htmlType: 'button' | 'submit' | 'reset';
+  handleClick?: (event: MouseEvent<HTMLButtonElement>) => void;
+  type?: 'primary' | 'default' | 'dashed' | 'text' | 'link';
+  isDisabled?: boolean;
+  size?: 'small' | 'large';
+  ghost?: boolean;
+  shape?: 'circle' | 'round';
+  color?: 'primary' | 'secondary';
+}
+
+export const Button = ({
+  children,
+  htmlType,
+  handleClick,
+  type,
+  isDisabled,
+  size,
+  ghost,
+  shape,
+  color,
+}: ButtonProps) => {
+  return (
+    <StyledButton
+      htmlType={htmlType}
+      onClick={handleClick}
+      type={type}
+      disabled={isDisabled}
+      size={size}
+      ghost={ghost}
+      shape={shape}
+      color={color}
+    >
+      {children ? <div>{children}</div> : null}
```

Comment:
Dlaczego w divie? Tak nie zadziała odpowiednio?
```suggestion
      {children}
```

Reply:
Bez wrappera react wyrzuca mi blad "Objects are not valid as a React child (found: object with keys {children}). If you meant to render a collection of children, use an array instead."